### PR TITLE
Create custom matchers for “valid symlink”.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Adds a custom matcher to check if a file is a symlink and its target exists.

### CODE REVIEW

File: Library/Homebrew/cask/spec/cask/artifact/binary_spec.rb [hunk 1]
```diff
@@ -20,7 +20,7 @@
     shutup do
       Hbc::Artifact::Binary.new(cask).install_phase
     end
-    expect(FileHelper.valid_alias?(expected_path)).to be true
+    expect(expected_path).to be_a_valid_symlink
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cask/spec/cask/artifact/binary_spec.rb]: Personally this seems harder to follow not given it's only used in two places.

Comment (reitermarkus) [inline Library/Homebrew/cask/spec/cask/artifact/binary_spec.rb]: I think this is more idiomatic in this case, and you are still able to look it up by name as you would be for `FileHelper.valid_alias?`.

Comment (MikeMcQuaid) [inline Library/Homebrew/cask/spec/cask/artifact/binary_spec.rb]: I'd argue for https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming) but don't feel strongly either way.

Comment (MikeMcQuaid) [inline Library/Homebrew/cask/spec/cask/artifact/binary_spec.rb]: Personally: yes. This PR is a refactoring for slightly shortening two lines of code which ends up adding more lines than it removes.

Comment (reitermarkus) [inline Library/Homebrew/cask/spec/cask/artifact/binary_spec.rb]: Ok, that makes sense.

**Pull request merged**